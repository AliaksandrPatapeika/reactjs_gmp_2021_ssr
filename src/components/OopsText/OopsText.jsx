import './OopsText.less';

import PropTypes from 'prop-types';
import React from 'react';

const OopsText = ({errorMessage}) => (
  <div className="oopsTextContainer">
    {errorMessage && (
    <div className="oopsTextErrorMessage">{errorMessage}</div>
    )}
    <div>Oops, something went wrong...</div>
    <div>We are doing our best to fix the issue</div>
  </div>
);

OopsText.propTypes = {
  errorMessage: PropTypes.string
};

OopsText.defaultProps = {
  errorMessage: ''
};

export default OopsText;
