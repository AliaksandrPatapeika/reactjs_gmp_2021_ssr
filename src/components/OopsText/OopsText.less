@import "../../less/variables";

.oopsTextContainer {
	position: absolute;
	display: flex;
	flex-direction: column;
	justify-content: center;
	width: 100%;
	height: 100%;
	padding: 40px;
	color: @red;
	font-size: 50px;
	font-family: @NetflixSans-Medium;
	text-align: center;

	.oopsTextErrorMessage {
		margin-bottom: 50px;
		color: @light-gray;
	}
}