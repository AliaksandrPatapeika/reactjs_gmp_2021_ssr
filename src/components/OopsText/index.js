import OopsText from './OopsText';

export default OopsText;
