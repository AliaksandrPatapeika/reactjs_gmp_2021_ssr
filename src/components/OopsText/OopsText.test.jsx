import React from 'react';
import renderer from 'react-test-renderer';

import OopsText from './OopsText';

describe('OopsText', () => {
  test('renders correctly', () => {
    const oopsText = renderer
      .create(<OopsText errorMessage="Error message" />)
      .toJSON();

    expect(oopsText).toMatchSnapshot();
  });
});
