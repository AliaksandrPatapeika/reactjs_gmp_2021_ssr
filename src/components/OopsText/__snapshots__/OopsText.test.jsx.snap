// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OopsText renders correctly 1`] = `
<div
  className="oopsTextContainer"
>
  <div
    className="oopsTextErrorMessage"
  >
    Error message
  </div>
  <div>
    Oops, something went wrong...
  </div>
  <div>
    We are doing our best to fix the issue
  </div>
</div>
`;
